datafeeder/src/main/java/org/georchestra/datafeeder/service/geonetwork/GeoNetworkRemoteService.java

```diff
@@ -0,0 +1,203 @@
+     * </code>
+     * </pre>
+     * 
+     * @param metadataId
+     * 
+     * @param metadataId
+     * 
+     * @param xmlRecordAsString
+     * @return
+     */
+    public GeoNetworkResponse publish(@NonNull String metadataId, @NonNull Supplier<String> xmlRecordAsString) {
+
+        final URL apiBaseURL = this.config.getApiUrl();
+        final String url = apiBaseURL.toString();
+        final String xmlRecord = xmlRecordAsString.get();
+
+        HttpHeaders reqHeaders = createAdditionalRequestHeaders();
+
+        GeoNetworkResponse response = client.putXmlRecord(url, reqHeaders, metadataId, xmlRecord);
+
+        HttpStatus statusCode = response.getStatus();
+        String statusText = response.getStatusText();
+
+        if (statusCode.is2xxSuccessful()) {
+            return response;
+        }
+        if (statusCode.is4xxClientError()) {
+            throw new IllegalArgumentException(
+                    "Error creating metadata record: " + statusText + " (" + response.getErrorResponseBody() + ")");
+        }
+        throw new RuntimeException(
+                "Error creating metadata record: " + statusText + " (" + response.getErrorResponseBody() + ")");
+    }
+
+    public void checkServiceAvailable() throws IOException {
+        final URL apiBaseURL = this.config.getApiUrl();
+        final String url = apiBaseURL.toString();
+
+        HttpHeaders reqHeaders = createAdditionalRequestHeaders();
+
+        client.checkServiceAvailable(url, reqHeaders);
+    }
+
+    public String getRecordById(@NonNull String metadataId) {
+
+        final URL apiBaseURL = this.config.getApiUrl();
+        final String url = apiBaseURL.toString();
+
+        HttpHeaders reqHeaders = createAdditionalRequestHeaders();
+
+        String response = client.getXmlRecord(url, reqHeaders, metadataId);
+
+        return response;
+    }
+
+    private HttpHeaders createAdditionalRequestHeaders() {
+        // Allow passing restricted headers
+        System.setProperty("sun.net.http.allowRestrictedHeaders", "true");
+
+        HttpHeaders reqHeaders = new HttpHeaders();
+        /// reqHeaders.set("Host", "localhost:28080");
+        // reqHeaders.set("Accept", "application/json");
+        // reqHeaders.set("Content-Type", "application/xml");
+        // reqHeaders.set("User-Agent", "curl/7.68.0");
+
+//		reqHeaders.set("Origin", "http://localhost:28080");
+//		reqHeaders.set("Referer", "http://localhost:28080");
+
+        reqHeaders.set("sec-proxy", "true");
+        reqHeaders.set("sec-username", "testadmin");
+        reqHeaders.set("sec-roles", "ROLE_GN_ADMIN");
+        reqHeaders.set("sec-org", "Datafeeder Test");
+        // This is odd, apparently any UUID works as XSRF token, and these two need to
+        // be set
+        reqHeaders.set("X-XSRF-TOKEN", "c9f33266-e242-4198-a18c-b01290dce5f1");
+        reqHeaders.set("Cookie", "XSRF-TOKEN=c9f33266-e242-4198-a18c-b01290dce5f1");
+        return reqHeaders;
```

Comment:
meaning that they have to be set, but whatever the value, they are not checked by GN ? weird indeed

Reply:
yup